### PR TITLE
docs(eslint-plugin): add script to generate the readme tables

### PR DESCRIPTION
**Pull request by G-Rath**
I created a new branch b/c it was easier than trying to rebase my existing branch.

Closes #1435

### CODE REVIEW

File: packages/eslint-plugin/tools/generate-rules-lists.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+const extensionRules = rulesDetails.filter(rule => rule.extendsBaseRule);
+
+let readme = fs.readFileSync(path.resolve(__dirname, '../README.md'), 'utf8');
+
+readme = updateRulesList('base', baseRules, readme);
+readme = updateRulesList('extension', extensionRules, readme);
+
+readme = prettier.format(readme, { parser: 'markdown' });
+
+fs.writeFileSync(path.resolve(__dirname, '../README.md'), readme, 'utf8');
```
Comment (armano2) [inline packages/eslint-plugin/tools/generate-rules-lists.ts]: before saving can you add prettier call, it should be something like this:
```ts
prettier.format(" 1", { parser: "markdown" });
```

or in package.json add execution of prettier on this file

```bash
prettier --write ./README.md
```

Comment (G-Rath) [reply packages/eslint-plugin/tools/generate-rules-lists.ts]: You mean like [this](https://github.com/typescript-eslint/typescript-eslint/pull/1524/files#diff-09b2350bad1cde9cfd0eefac53f95aa6R127)? 😜

**Comment by G-Rath** [comment]
> Cannot find module './lint-output.json' from 'test.js'

That's an interesting integration test failure that I'm pretty sure I didn't cause?

File: packages/eslint-plugin/tools/generate-rules-lists.ts (hunk 1 above)
Comment (armano2) [reply packages/eslint-plugin/tools/generate-rules-lists.ts]: i'm getting blind

**Comment by armano2** [comment]
i restarted build, for unknown reason in integration tests there was no typescript installed

**Pull request merged**